### PR TITLE
feat(aom): add aom service_discovery_rule resource

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add aom service_discovery_rule resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
add aom service_discovery_rule resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/aom' TESTARGS='-run TestAccAOMServiceDiscoveryRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run TestAccAOMServiceDiscoveryRule_basic -timeout 360m -parallel 4
=== RUN   TestAccAOMServiceDiscoveryRule_basic
=== PAUSE TestAccAOMServiceDiscoveryRule_basic
=== CONT  TestAccAOMServiceDiscoveryRule_basic
--- PASS: TestAccAOMServiceDiscoveryRule_basic (93.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       94.044s
```

### CODE REVIEW

File: docs/resources/aom_service_discovery_rule.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Manages a AOM service discovery rule resource within HuaweiCloud.
+
+## Example Usage
+
+### Basic example
+
+```hcl
+resource "huaweicloud_aom_service_discovery_rule" "discovery_rule" {
+  rule_name              = "test-rule"
```
Comment (niuzhenguo) [inline docs/resources/aom_service_discovery_rule.md]: how about just naming it `name` as it's an atrribute of rule resource.

File: docs/resources/aom_service_discovery_rule.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  priovrity               = 9999
```
Comment (niuzhenguo) [inline docs/resources/aom_service_discovery_rule.md]: s/priovrity/priority

File: docs/resources/aom_service_discovery_rule.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+* `log_file_fix` - (Required, List) Specifies the log file suffix. This is a list of strings.
+  Values can be: log, trace, and out.
+
+* `app_type` - (Required, String) Specifies the service type, which is used only for rule classification and UI display.
+  You can enter any field. For example, enter Java or Python by technology stack, or enter collector or database by function.
+
+* `detect_log_enabled` - (Optional, Bool) Specifies the whether to enable log collection. The default value is true.
+
+* `priovrity` - (Optional, Int) Specifies the rule priority. Value range: 1 to 9999. The default value is 9999.
```
Comment (niuzhenguo) [inline docs/resources/aom_service_discovery_rule.md]: s/priovrity/priority

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  }
+}
+`, rName)
+}
+
+func testAOMServiceDiscoveryRule_update(rName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_aom_service_discovery_rule" "test" {
+  rule_name              = "%s"
+  priovrity              = 9998
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go]: s/priovrity/priority

File: huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+			"app_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"detect_log_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"priovrity": {
```
Comment (niuzhenguo) [inline huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go]: s/priovrity/priority

File: docs/resources/aom_service_discovery_rule.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md]: sure

File: docs/resources/aom_service_discovery_rule.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md]: fixed

File: docs/resources/aom_service_discovery_rule.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md]: fixed

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go]: fixed

File: huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go]: fixed

File: docs/resources/aom_service_discovery_rule.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the service discovery rule resource. If omitted,
+  the provider-level region will be used. Changing this creates a new resource.
+
+* `name` - (Required, String) Specifies the rule name, which contains a maximum of 64 characters. It must start
```
Comment (hmbig2) [inline docs/resources/aom_service_discovery_rule.md]: min is 4, pls add

File: huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
```
Comment (hmbig2) [inline huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go]: s/validation.StringLenBetween(1, 64)/validation.StringLenBetween(4, 63),

File: docs/resources/aom_service_discovery_rule.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+  the command are extracted. If the value of `name_type` is **env**, `args` is in the format of ["aa"], indicating that
+  the environment variable named aa is extracted. If the value of `name_type` is **str**, `args` is in the format of
+  ["fix"], indicating that the application name is suffixed with fix. If the value of `name_type` is **cmdLineHash**,
+  `args` is in the format of ["0001"] and `value` is in the format of ["ser"], indicating that the application name is
+  ser when the startup command is 0001. The [object](#basic_name_rule_object) structure is documented below.
+
+<a name="basic_name_rule_object"></a>
+The `service_name_rule` block and `application_name_rule` block support:
+
+* `name_type` - (Required, String) Specifies the value type. The value can be **cmdLineHash**, **cmdLine**, **env**
```
Comment (hmbig2) [inline docs/resources/aom_service_discovery_rule.md:119]: pls add validate to schema

File: docs/resources/aom_service_discovery_rule.md (hunk 6 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md]: fixed

File: huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/aom/resource_huaweicloud_aom_service_discovery_rule.go]: fixed

File: docs/resources/aom_service_discovery_rule.md (hunk 8 above)
Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md:119]: fixed

**Comment by Jason-Zhang9309** [comment]
make testacc TEST='./huaweicloud/services/acceptance/aom' TESTARGS='-run TestAccAOMServiceDiscoveryRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run TestAccAOMServiceDiscoveryRule_basic -timeout 360m -parallel 4
=== RUN   TestAccAOMServiceDiscoveryRule_basic
=== PAUSE TestAccAOMServiceDiscoveryRule_basic
=== CONT  TestAccAOMServiceDiscoveryRule_basic
--- PASS: TestAccAOMServiceDiscoveryRule_basic (23.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       23.157s

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+  discovery_rules {
+    check_content = ["python"]
+    check_mode    = "contain"
+    check_type    = "cmdLine"
+  }
+
+  log_path_rules {
+    name_type = "cmdLineHash"
+    args      = ["python"]
+	value     = ["/tmp/log"]
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go]: format error

File: docs/resources/aom_service_discovery_rule.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+  to **equals**. `check_content` is in the format of ["hostId1","hostId2"], indicating that the rule takes effect only
+  on specified nodes. If no nodes are specified, the rule applies to all nodes of the project.
+  + `check_type` - (Required, String) Specifies the match type. The values can be **cmdLine**, **env** and **scope**.
+  + `check_mode` - (Required, String) Specifies the match condition. The values can be **contain** and **equals**.
+  + `check_content` - (Required, List) Specifies the matched value. This is a list of strings.
+
+* `name_rules` - (Required, List) Specifies the naming rules for discovered services and applications.
+  The [object](#name_rules_object) structure is documented below.
+
+* `log_file_fix` - (Required, List) Specifies the log file suffix. This is a list of strings.
```
Comment (ShiChangkuo) [inline docs/resources/aom_service_discovery_rule.md]: is `log_file_suffix` more appropriate?

Comment (Jason-Zhang9309) [reply docs/resources/aom_service_discovery_rule.md]: sure

File: huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go (hunk 9 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_service_discovery_rule_test.go]: fixed

**Pull request merged**